MobileWallet/Screens/Home/TransactionHistory/TransactionsTableViewController.swift

```diff
@@ -71,18 +71,31 @@ class TransactionsTableViewController: UITableViewController {
         }
     }
 
-    private var tempRefreshTimer = Timer()
-
     override func viewDidLoad() {
         super.viewDidLoad()
         viewSetup()
         tableView.register(UINib(nibName: CELL_IDENTIFIER, bundle: nil), forCellReuseIdentifier: CELL_IDENTIFIER)
         refreshTransactionControl.attributedTitle = NSAttributedString(string: "Pull to refresh") //TODO local
-        refreshTransactionControl.addTarget(self, action: #selector(refreshTransactions(_:)), for: .valueChanged)
-        //refreshControl = refreshTransactionControl
+        refreshTransactionControl.addTarget(self, action: #selector(refreshPullTransactions(_:)), for: .valueChanged)
 
         self.refreshTable()
-        tempRefreshTimer = Timer.scheduledTimer(timeInterval: 2, target: self, selector: #selector(self.refreshTable), userInfo: nil, repeats: true)
+
+        TariEventBus.onMainThread(self, eventType: .receievedTransaction) { (result) in
+            if let tx: PendingInboundTransaction = result?.object as? PendingInboundTransaction {
+                print("New transaction receieved ", tx.id.0)
+                //TODO animate in the new TX instead of just refreshing the table below
+            }
+        }
+
+        TariEventBus.onMainThread(self, eventType: .transactionListUpdate) { (_) in
+            self.refreshTable()
```

Comment:
same as above
TariEventBus.onMainThread(self, eventType: . transactionListUpdate) { [weak self] (_) in guard let self = self else { return } self.refreshBalance()